src/Nethermind/Nethermind.AuRa.Test/Transactions/PermissionTxComparerFactoryTests.cs

```diff
@@ -264,8 +265,14 @@ public void order_is_correct(Func<IEnumerable<Transaction>, IEnumerable<Transact
             transactions = transactionSelect?.Invoke(transactions) ?? transactions;
             expectation = transactionSelect?.Invoke(expectation) ?? expectation;
             
-            var selectionStrategy = new PermissionTxPoolOrderingStrategy(sendersWhitelist, priorities);
-            var orderedTransactions = selectionStrategy.Order(blockHeader, transactions).ToArray();
+            var comparerFactory = new PermissionTxComparerFactory(sendersWhitelist, priorities);
+            var comparer = comparerFactory.CreateComparer(blockHeader);
+            var txBySender = transactions.GroupBy(t => t.SenderAddress)
+                .ToDictionary(
+                    g => g.Key, 
+                    g => g.OrderBy(t => t, new TxPool.TxPool.NonceCompositeComparer(comparer)).ToArray());
```

Comment:
renive the comprer from the TxPool class, comparer should be reusable across multiple TxPool implementations